src/server/journal/streamer.cc

```diff
@@ -92,6 +92,13 @@ void RestoreStreamer::Start(io::Sink* dest) {
   });
 }
 
+void RestoreStreamer::SendFinalize() {
+  VLOG(2) << "DFLYMIGRATE FINALIZE for " << sync_id_ << " : " << db_slice_->shard_id();
+  WriteCommand(make_pair("DFLYMIGRATE", ArgSlice{"FINALIZE", absl::StrCat(sync_id_),
```

Comment:
Sending new opcode instead of a command can simplify the flow very much. here and for full sync cut as well.
Now you need to implement the command dflymigrate finlize sync_id shard_id
instead if you send ofcode FIN than inside ClusterShardMigration::FullSyncShardFb you will have a while loop which you will break from when you get the FIN opcode.